activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -469,7 +476,12 @@ def find_one(id)
           MSG
         end
 
-        relation = where(primary_key => id)
+        relation = if klass.composite_primary_key?
+          where(Array(primary_key).zip(Array(id)).to_h)
```

Comment:
There doesn't seem to be any tests where these aren't arrays. Is it actually possible for these to not be an array? If not we should remove the `Array` around them to end up with `where(primary_key.zip(id).to_h)`

Reply:
You are right, this wasn't needed. It was added to handle both single-column and composite pks but then we decided to branch